tests/test_unittest/test_cli.py

```diff
@@ -0,0 +1,94 @@
+"""Unit tests for the command line interface (CLI)."""
+import re
+
+import pytest
+from click.testing import CliRunner
+
+from gt4py import cli
+from gt4py import backend
+
+
+@pytest.fixture
+def clirunner():
+    """Make the CLiRunner instance conveniently available."""
+    yield CliRunner()
+
+
+@pytest.fixture(params=backend.REGISTRY.keys())
+def backend_name(request):
+    """Parametrize by backend name."""
+    yield request.param
+
+
+class NonCliBackend(backend.BaseBackend):
+    """This represents a non-CLI-enabled backend."""
+
+    name = "nocli"
+
+    def generate(*args, **kwargs):
+        pass
+
+    def load(*args, **kwargs):
+        pass
+
+
+@pytest.fixture
+def nocli_backend():
+    """Temporarily register the nocli backend."""
+    backend.register(NonCliBackend)
+    yield
+    backend.REGISTRY.pop("nocli")
+
+
+BACKEND_ROW_BY_NAME = {
```

Comment:
I think the name of this dict is a bit misleading since it is actually storing a RE to match the backend names, not backend names themselves.

Reply:
it stores row patterns *by* name